markdown/treeprocessors.py

```diff
@@ -271,7 +271,12 @@ def __applyPattern(
             String with placeholders instead of `ElementTree` elements.
 
         """
-        new_style = isinstance(pattern, inlinepatterns.InlineProcessor)
+        if isinstance(pattern, inlinepatterns.InlineProcessor):
+            new_style = True
+            new_pattern = pattern
+        else:  # pragma: no cover
+            new_style = False
+            legacy_pattern = pattern
```

Comment:
Again this (and all the associated changes in this file that follow) is removing the dynamic nature of python.

Reply:
OK let's just ignore-comment everything instead, I don't know

Reply:
A cool thing to do here would be to refactor this method into 2 methods. Despite some code duplication, it's actually less error-prone going forward, and even provides a clearer deprecation for old-style patterns. (Also only one `# pragma: no cover` comment now, rather than per-case ones 😊

This is demonstrated here: https://github.com/oprypin/python-markdown/commit/1c047f02bc86edbdd5a0eee69aaff37730a080ed?diff=split

Reply:
> OK let's just ignore-comment everything instead, I don't know

Actually, as per my [other comment](https://github.com/Python-Markdown/markdown/pull/1402#issuecomment-1792588190), not don't just ignore-comment everything.

> A cool thing to do here would be to refactor this method into 2 methods.

Perhaps, but that should be addresses separately.